csharp-basic-topics/PerformSubstringSearchWithContainsMethod/PerformSubstringSearchWithContainsMethod/SubstringSearch.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Text.RegularExpressions;
+
+namespace PerformSubstringSearchWithContainsMethod
+{
+    public class SubstringSearch
+    {
+        public static void Main(string[] args)
+        {
+            var sourceString = "Code Maze";
+            var substringToSearch = "maze";
+            var separator = ' ';
+
+            // String Contains Method Search
+            StringContainsSubstringWithNoParameter(sourceString, substringToSearch);
+            StringContainsSubstringWithOrdinalStringComparisonParameter(sourceString, substringToSearch);
+            StringContainsSubstringWithOrdinalIgnoreCaseStringComparisonParameter(sourceString, substringToSearch);
+
+            // String IndexOf Method Search
+            StringIndexOfMethodSearchWithNoParameter(sourceString, substringToSearch);
+            StringIndexOfMethodSearchWithOrdinalStringComparisonParameter(sourceString, substringToSearch);
+            StringIndexOfMethodSearchWithOrdinalIgnoreCaseStringComparisonParameter(sourceString, substringToSearch);
+
+            // Regular Expression Search
+            RegularExpressionSearchWithNoParameter(sourceString, substringToSearch);
+            RegularExpressionSearchWithRegexOptionsNoneParameter(sourceString, substringToSearch);
+            RegularExpressionSearchWithRegexOptionsIgnoreCaseParameter(sourceString, substringToSearch);
+
+            // Linq With String Equals Method Search
+            LinqWithStringEqualsMethodSearchWithNoParameter(sourceString, substringToSearch, separator);
+            LinqWithStringEqualsMethodSearchWithOrdinalStringComparisonParameter(sourceString, substringToSearch, separator);
+            LinqWithStringEqualsMethodSearchWithOrdinalIgnoreCaseStringComparisonParameter(sourceString, substringToSearch, separator);
+        }
```

Comment:
This code is basically the console app part and should be split into its own file. We can just add a Program.cs and drop this in there. Since this project targets .NET 7.0, we can drop the `public static void Main` part and just drop in the heart of the method (using top level statements)